From 3c59db57716e09fc5a38d08621ac25eef69b8fc4 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Fri, 21 Dec 2012 16:48:13 -0800
Subject: [PATCH 132/518] netdev/phy/of: Handle nexus Ethernet PHY devices

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: f6e59de722050a0aa9339b518cc86c3cbc21b066
Description:

Some multi-phy devices have resources that are global between all of the
PHYs on the same device such as the Vitesse vsc8488.  In this case each
individual PHY is contained within a phy nexus device.

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
Signed-off-by: Leonid Rosenboim <lrosenboim@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/of/of_mdio.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/of/of_mdio.c b/drivers/of/of_mdio.c
index d5a57a9..5151e46 100644
--- a/drivers/of/of_mdio.c
+++ b/drivers/of/of_mdio.c
@@ -17,6 +17,7 @@
 #include <linux/of.h>
 #include <linux/of_irq.h>
 #include <linux/of_mdio.h>
+#include <linux/of_platform.h>
 #include <linux/module.h>
 
 MODULE_AUTHOR("Grant Likely <grant.likely@secretlab.ca>");
@@ -73,6 +74,11 @@ int of_mdiobus_register(struct mii_bus *mdio, struct device_node *np)
 			continue;
 		}
 
+		if (of_device_is_compatible(child, "ethernet-phy-nexus")) {
+			of_platform_device_create(child, NULL, &mdio->dev);
+			continue;
+		}
+
 		if (mdio->irq) {
 			mdio->irq[addr] = irq_of_parse_and_map(child, 0);
 			if (!mdio->irq[addr])
-- 
1.9.1

